app/models/concerns/person_concern.rb

```diff
@@ -8,7 +8,7 @@ module PersonConcern
     validates :email, format: { with: /\A.+@.+\..+\z/ }, allow_blank: true
 
     def full_name
-      "#{first_name} #{last_name}"
+      [first_name, last_name].compact.join(' ')
```

Comment:
This specific change should prevent the random failures in circle-ci when the mocked data has a composed `last_name`. (e.g. “Da Silva”)

Reply:
Hmm je ne suis pas sur que ça soit la bonne solution au problème observé, car dans le cas où tu as un `last_name` composé, ces deux méthodes sont sensées avoir exactement le même comportement.

À la limite, la seule différence entre l'avant et l'après, c'est que tu gères de façon différente les cas où `first_name` et/ou `last_name` est `nil`.

Donc pour moi cette modification n'est pas sensée reglér le problème que tu décris (mais je peux rater un truc)

Reply:
C’est effectivement un workaround pour un problème plus sérieux.

Le cas qui casse dans les tests, c’est que dans les données mockées, `first_name` est nil et `last_name` a une valeur aléatoire, parfois un nom composé. `nil, "Da Silva"` donne comme `full_name` `" Da Silva"`, avec une espace au début.

Le problème plus profond est lié à la méthode `full_name=` dans `Contact`, qui `split(' ')` pour deviner `first_name` et `last_name`. Concrètement, l’app n’a pas besoin de `first_name` et `last_name` distincts. Je transformerais bien tout ça en `name` tout court.

Reply:
Clairement. Je vais revenir à la version précédente, le `compact.join()` règle le test qui échoue aléatoirement, mais n’a pas de sens à la lecture du code.